src/core/server/http/auth_state_storage.ts

```diff
@@ -45,12 +45,12 @@ export type IsAuthenticated = (request: KibanaRequest) => boolean;
 
 /** @internal */
 export class AuthStateStorage {
-  private readonly storage = new WeakMap<LegacyRequest, unknown>();
+  private readonly storage = new WeakMap<Request, unknown>();
   constructor(private readonly canBeAuthenticated: () => boolean) {}
-  public set = (request: KibanaRequest | LegacyRequest, state: unknown) => {
+  public set = (request: KibanaRequest | Request, state: unknown) => {
```

Comment:
```suggestion
  public set = (request: KibanaRequest, state: unknown) => {
```

Reply:
We still need to allow `Hapi.Request` as the request param type for the [auth scheme](https://github.com/elastic/kibana/blob/main/src/core/server/http/http_server.ts#L436-L455). If we did want to revisit that, we'd need to do some more refactoring.